src/block/block_ckpt.c

```diff
@@ -516,11 +563,14 @@ __ckpt_add_blk_mods_ext(WT_SESSION_IMPL *session, WT_CKPT *ckptbase, WT_BLOCK_CK
             continue;
 
         if (ci->alloc.offset != WT_BLOCK_INVALID_OFFSET)
-            WT_RET(__ckpt_add_blkmod_entry(session, blk_mod, ci->alloc.offset, ci->alloc.size));
-        if (ci->discard.offset != WT_BLOCK_INVALID_OFFSET)
-            WT_RET(__ckpt_add_blkmod_entry(session, blk_mod, ci->discard.offset, ci->discard.size));
+            WT_RET(__ckpt_mod_blkmod_entry(
+              session, blk_mod, ci->alloc.offset, ci->alloc.size, true, "alloc"));
         if (ci->avail.offset != WT_BLOCK_INVALID_OFFSET)
-            WT_RET(__ckpt_add_blkmod_entry(session, blk_mod, ci->avail.offset, ci->avail.size));
+            WT_RET(__ckpt_mod_blkmod_entry(
+              session, blk_mod, ci->avail.offset, ci->avail.size, true, "avail"));
+        if (ci->discard.offset != WT_BLOCK_INVALID_OFFSET)
+            WT_RET(__ckpt_mod_blkmod_entry(
+              session, blk_mod, ci->discard.offset, ci->discard.size, true, "discard"));
```

Comment:
This code adds new extent list blocks to the incremental bitmaps. I.e., when we change the extent lists for an old checkpoint and rewrite its extent lists, we use this code to ensure that the newly allocated blocks holding those extent lists get included in the incremental block lists.

The other side of this is that we also free the blocks ([here](https://github.com/wiredtiger/wiredtiger/blob/289e27631710c9c5e4e7cd67f4e85e56b966fbf2/src/block/block_ckpt.c#L770-L771)) that held the old extent lists. Now that we are tracking discarded blocks, it would be nice to remove those from our bitmaps as well.

Technically, this isn't required for correctness, and it's probably only a small amount of extra data we might wind up backing up. But it would be nice to do.

Reply:
Adding it in where we free the blocks is slightly complicated due to wanting to not clear bits unless the free was already successful and the recent change I had to put in not to clear bits when named checkpoints exist. It would require a bunch of refactoring. I'll add a comment there and create a followup ticket.